src/components/Menus/Menu.js

```diff
@@ -0,0 +1,204 @@
+import React             from 'react'
+import PropTypes         from 'prop-types'
+
+const styles = {
+  borderRadius: '4px',
+  backgroundColor: '#ffffff',
+  boxShadow: '0 3px 10px 0 rgba(0, 0, 0, 0.16), 0 3px 10px 0 rgba(0, 0, 0, 0.23)',
+  display: 'inline-block',
+  padding: '8px 0',
+  overflowY: 'auto',
+  userSelect: 'none',
+  outline: 'none'
+}
+
+class Menu extends React.Component {
+  static propTypes = {
+    /** aria-hidden HTML attribute */
+    ariaHidden: PropTypes.bool,
+    /** MenuItems or Divider */
+    children: PropTypes.node.isRequired,
+    /** The maximum height of the Menu when it is displayed */
+    maxHeight: PropTypes.number,
+    /** Callback function fired on key down */
+    onKeyDown: PropTypes.func,
+    /** Callback function fired when a MenuItem is selected */
+    onSelect: PropTypes.func,
+    /** Callback function fired when a Menu is blurred */
+    onBlur: PropTypes.func,
+    /** Role HTML attribute */
+    role: PropTypes.string,
+    /** Customize style of menu parent */
+    style: PropTypes.shape({})
+  }
+
+  static defaultProps = {
+    role: 'menu',
+    maxHeight: 500,
+    ariaHidden: false,
+    style: {},
+  }
+
+  state = {
+    currentTabIndex: null
+  }
+
+  _handleClick = (option, index) => {
+    const { onSelect } = this.props
+
+    onSelect && onSelect(option, index)
```

Comment:
I think we are trying to implement a pattern where we have a consistent/predictable callback signature:
```
callback(event, {...props, ...state })
```

Reply:
This specific callback needs to pass the selected option. Would this just be appended to the end of the arguments? Not sure i like this pattern.

Reply:
Well it seems like this is actually the wrong place to put the click handler, `Menu` should just pass down `onSelect` to `MenuItem` and `MenuItem` should pass it's props to the callback. Then you have the label and value within the properties argument:
```
class MenuItem {
  handleClick = (e) => {
    nSelect && onSelect(e, this.props)
   }
}
```

Reply:
For something like an input element you would extract the value from the event and pass it along with the props:
```
class SomeSnacksCustomInput {
  handleClick = (e) => {
    const { value } = e.target
    onSelect && onSelect(e, {...this.props, value })
   }
}
```

Reply:
We don't have to go with this specific pattern, but I think it's very important we standardize these within `Snacks` to maintain standards and predictability.  FWIW I've used this pattern in large scale FE applications in the past and it allows you to create more abstract handlers if you have a know the shape of the payload.